### PR TITLE
Make `Style/GuardClause` a bit more lenient when the replacement would make the code more verbose

### PR DESCRIPTION
**Pull request by dvandersluis**
We currently have an exception in `Style/GuardClause` when the suggested replacement would be too long.

```ruby
def test
  if something && something_that_makes_the_guard_clause_too_long_to_fit_on_one_line
    work
  end
end
```

RuboCop suggests replacing this with:

```ruby
def test
  unless something && something_that_makes_the_guard_clause_too_long_to_fit_on_one_line
    return
  end
  work
end
```

However, this makes the code more verbose (adds a line that doesn't really add any benefit) and harder to understand (since this only happens when the line is too long, it's a safe assumption that the condition is long, and flipping it with `unless` adds cognitive overhead).

This exception is not covered in the [style guide](https://rubystyle.guide/#no-nested-conditionals) either, so it should be "non-breaking".

My proposal here is that if the guard clause is too line but what is inside the `if` is trivial (a single statement, and not a nested if), it will not be registered as an offense.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/guard_clause.rb [hunk 1]
```diff
@@ -134,14 +134,12 @@ def consecutive_conditionals?(parent, node)
           end
         end
 
-        def opposite_keyword(node)
-          node.if? ? 'unless' : 'if'
-        end
```
Comment (dvandersluis) [inline lib/rubocop/cop/style/guard_clause.rb:139]: This is not strictly related to my change but I noticed it could be replaced with `node.inverse_keyword`.

https://github.com/rubocop/rubocop-ast/blob/816dfe7f2ca4e92c7eda226a9e8b44aa9fa81e81/lib/rubocop/ast/node/if_node.rb#L61-L73

**Comment by bbatsov** [comment]
That's an improvement for sure, but overall I think that's one of those cops that need more work and have to be relaxed a bit to focus on cases where the guard clause really improves something.
